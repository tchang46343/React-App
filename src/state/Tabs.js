import React from "react";
import "bootstrap/dist/css/bootstrap.css";

class Tabs extends React.Component {
  static defaultProps = { tabs: [] };
  state = {
    currentTabIndex: 0
  };
  //   render() {
  //     const buttons = this.props.tabs.map((tab, index) => (
  //       <button key={index}>{tab.name}</button>
  //     ));
  //     const currentTab = this.props.tabs[0];
  //     return (
  //       <div>
  //         {buttons}
  //         {this.props.tabs.length && (
  //           <div className="content">{currentTab.content}</div>
  //         )}
  //       </div>
  //     );
  //   }

  handleButtonClick = index => {
    this.setState({ currentTabIndex: index });
    // console.log("button clicked!", { index });
  };

  renderButtons() {
    return this.props.tabs.map((tab, index) => (
      <button
        // variant="primary"
        key={index}
        onClick={() => this.handleButtonClick(index)}
      >
        {tab.name}
      </button>
    ));
  }

  renderContent() {
    const currentTab = this.props.tabs[this.state.currentTabIndex];
    return <div className="content">{currentTab.content}</div>;
  }

  render() {
    return (
      <div>
        {this.renderButtons()}
        {!!this.props.tabs.length && this.renderContent()}
      </div>
    );
  }
}

export default Tabs;
