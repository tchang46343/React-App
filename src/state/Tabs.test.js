import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Tabs from "./Tabs";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe(`Tabs Component`, () => {
  const tabsProp = [
    {
      name: "About us",
      content:
        "At InnovateX we provide the highest quality products for the aero Space Industry."
    },
    {
      name: "Services",
      content:
        "We provide aerospace level metal fabrication and computer transitors"
    },
    { name: "Contact", content: "InnovateX@gmail.com" }
  ];

  it("render the first tab by default", () => {
    const tree = renderer.create(<Tabs tabs={tabsProp} />);
    expect(tree).toMatchSnapshot();
  });

  it("renders without errors", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Tabs />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders empty given no tabs", () => {
    const wrapper = shallow(<Tabs />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("closes the first tab and opens any clicked tab", () => {
    const wrapper = shallow(<Tabs tabs={tabsProp} />);
  });
});
