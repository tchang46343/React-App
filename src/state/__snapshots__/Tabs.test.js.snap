// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Component render the first tab by default 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    About us
  </button>
  <button
    onClick={[Function]}
  >
    Services
  </button>
  <button
    onClick={[Function]}
  >
    Contact
  </button>
  <div
    className="content"
  >
    At InnovateX we provide the highest quality products for the aero Space Industry.
  </div>
</div>
`;

exports[`Tabs Component renders empty given no tabs 1`] = `<div />`;
